### PR TITLE
Add consensus_channel fetch function

### PR DESCRIPTION
**Pull request by NiloCK**
Incremental toward #420.

This PR is a step toward side-by-side functioning of the existing `TwoPartyLedger` channels and incoming `ConsensusChannel` implementations. It provides the virtualfund objective with the means to populate its consensus channels from the store.

---

#### Code quality

- [x] I have written clear commit messages
- [x] I have performed a self-review of my own code
- [x] This change does not have an unduly wide scope
- [x] I have separated logic changes from refactor changes (formatting, renames, etc.)
- [x] I have commented my code wherever necessary (can be 0)
- [ ] I have added tests that prove my fix is effective or that my feature works, if necessary
  - not necessary yet
- [x] I have added comprehensive [godoc](https://go.dev/blog/godoc) comments 

#### Project management

- [x] I have assigned myself to this PR
- [x] I have linked the appropriate github issue
- [x] I have assigned this PR to the appropriate GitHub project
- [x] I have assigned this PR to the appropriate GitHub Milestone

### CODE REVIEW

File: protocols/virtualfund/virtualfund.go [hunk 1]
```diff
@@ -496,9 +496,17 @@ func (o *Objective) isBob() bool {
 // GetTwoPartyLedgerFunction specifies a function that can be used to retreive ledgers from a store.
 type GetTwoPartyLedgerFunction func(firstParty types.Address, secondParty types.Address) (ledger *channel.TwoPartyLedger, ok bool)
 
+// todo: #420 assume name and godoc from GetTwoPartyLedgerFunction
+type GetTwoPartyConsensusLedgerFunction func(leader, follower types.Address) (ledger *consensus_channel.ConsensusChannel, ok bool)
```
Comment (geoknee) [inline protocols/virtualfund/virtualfund.go:500]: Nit: 
```suggestion
type GetTwoPartyConsensusChannelFunction func(leader, follower types.Address) (ledger *consensus_channel.ConsensusChannel, ok bool)
```

and so on elsewhere.

File: protocols/virtualfund/virtualfund.go [hunk 2]
```diff
@@ -523,15 +533,36 @@ func ConstructObjectiveFromMessage(m protocols.Message, myAddress types.Address,
...
 		if !ok {
 			return Objective{}, fmt.Errorf("could not find a right ledger channel between %v and %v", intermediary, bob)
 		}
+
+		rightC, _ = getTwoPartyConsensusLedger(intermediary, bob)
+		// if !ok {
+		// 	// todo: #420 handle, after consensus_channel lifecycle is defined & channels present
+		// 	return Objective{}, fmt.Errorf("could not find a right ledger channel between %v and %v", intermediary, bob)
+		// }
+
```
Comment (geoknee) [inline protocols/virtualfund/virtualfund.go:565]: I think we are very close to being able to uncomment this, so I would suggest we do not merge this until that is possible?

**Review by geoknee — APPROVED** [review]
Can we put in some comments indicating that we do not expect to have _both_ types of channel in place in the long term?

**Comment by NiloCK** [comment]
closing:

changes from this PR were merged to `main` via #473

**Pull request closed without merging**